app/assets/javascripts/admin/admin_rest_api.js

```diff
@@ -1,58 +1,57 @@
 // @flow
 import _ from "lodash";
-import Request from "libs/request";
-import Toast from "libs/toast";
-import type { Message } from "libs/toast";
-import * as Utils from "libs/utils";
-import window, { location } from "libs/window";
-import messages from "messages";
-import { parseProtoTracing } from "oxalis/model/helpers/proto_helpers";
-import type {
-  APIUser,
-  APIScript,
-  APIScriptCreator,
-  APIScriptUpdater,
-  APITaskType,
-  APITeam,
-  APIProject,
-  APIProjectWithAssignments,
-  APIProjectCreator,
-  APIProjectUpdater,
-  APITask,
-  APIAnnotation,
-  APIAnnotationWithTask,
-  APIDataStore,
-  APITracingStore,
-  DatasetConfig,
-  APIDatasetId,
-  APIDataset,
-  APIMaybeUnimportedDataset,
-  APIDataSource,
-  APIDataSourceWithMessages,
-  APITimeInterval,
-  APIUserLoggedTime,
-  APITimeTracking,
-  APIProjectProgressReport,
-  APIOpenTasksReport,
-  APIBuildInfo,
-  APITracingType,
-  APIFeatureToggles,
-  APIOrganization,
-  ServerTracing,
-  APIActiveUser,
-  HybridServerTracing,
-  ServerSkeletonTracing,
-  ServerVolumeTracing,
-  APIAnnotationTypeCompact,
-  APIUpdateActionBatch,
-  ExperienceDomainList,
+
+import {
+  type APIActiveUser,
+  type APIAnnotation,
+  type APIAnnotationTypeCompact,
+  type APIAnnotationWithTask,
+  type APIBuildInfo,
+  type APIDataSource,
+  type APIDataSourceWithMessages,
+  type APIDataStore,
+  type APIDataset,
+  type APIDatasetId,
+  type APIFeatureToggles,
+  type APIMaybeUnimportedDataset,
+  type APIOpenTasksReport,
+  type APIOrganization,
+  type APIProject,
+  type APIProjectCreator,
+  type APIProjectProgressReport,
+  type APIProjectUpdater,
+  type APIProjectWithAssignments,
+  type APIScript,
+  type APIScriptCreator,
+  type APIScriptUpdater,
+  type APITask,
+  type APITaskType,
+  type APITeam,
+  type APITimeInterval,
+  type APITimeTracking,
+  type APITracingStore,
+  type APITracingType,
+  APITracingTypeEnum,
+  type APIUpdateActionBatch,
+  type APIUser,
+  type APIUserLoggedTime,
+  type DatasetConfig,
+  type ExperienceDomainList,
+  type HybridServerTracing,
+  type ServerSkeletonTracing,
+  type ServerTracing,
+  type ServerVolumeTracing,
 } from "admin/api_flow_types";
-import { APITracingTypeEnum } from "admin/api_flow_types";
-import type { QueryObject } from "admin/task/task_search_form";
-import type { NewTask, TaskCreationResponse } from "admin/task/task_create_bulk_view";
 import type { DatasetConfiguration } from "oxalis/store";
-import type { RequestOptions } from "libs/request";
+import type { NewTask, TaskCreationResponse } from "admin/task/task_create_bulk_view";
+import type { QueryObject } from "admin/task/task_search_form";
```

Comment:
Ah, so if it is **only types** from one imported file, importjs will put the `type` in front of the `{ ... }`. In that case, no unnecessary imports should be added by importjs, then :)

Reply:
Yeah, I figured that out too after a while :) The only thing which doesn't work well are * imports. E.g., "import * as Utils from ..." cannot be automatically inserted and is also screwed up when using "fix imports" in a file which already does so -.-
However, we only used these in very few places, which is why I rewrote the parts to use more precise imports (arguably better anyway). I'll file an issue at import-js nevertheless.